src/main/flight/mixer_tricopter.c

```diff
@@ -56,278 +56,228 @@
-        if ((angle >= minAngle) && (angle <= maxAngle))
-        {
+        if ((angle >= minAngle) && (angle <= maxAngle)) {
             maxNegForce = MIN(yawForceCurve[i], maxNegForce);
             maxPosForce = MAX(yawForceCurve[i], maxPosForce);
         }
         angle += 10;
     }
-
-    tailServoMaxYawForce = MIN(ABS(maxNegForce), ABS(maxPosForce));
+    tailServo.maxYawForce = MIN(ABS(maxNegForce), ABS(maxPosForce));
 }
 
-uint16_t triGetCurrentServoAngle()
+uint16_t triGetCurrentServoAngle(void)
 {
-    return tailServoAngle;
+    return tailServo.angle;
 }
 
 static uint16_t getLinearServoValue(servoParam_t *servoConf, int16_t constrainedPIDOutput)
 {
-    const int32_t linearYawForceAtValue = tailServoMaxYawForce * constrainedPIDOutput / TRI_YAW_FORCE_PRECISION;
-    const int16_t correctedAngle = getAngleFromYawCurveAtForce(linearYawForceAtValue);
-    return getServoValueAtAngle(servoConf, correctedAngle);
+    const int32_t linearYawForceAtValue = tailServo.maxYawForce * constrainedPIDOutput / TRI_YAW_FORCE_PRECISION;
+    const int16_t correctedAngle = getAngleFromYawForceCurve(linearYawForceAtValue);
+    const uint16_t linearServoValue = getServoValueAtAngle(servoConf, correctedAngle);
+
+    return linearServoValue;
 }
 
 static uint16_t getNormalServoValue(servoParam_t *servoConf, int16_t constrainedPIDOutput)
 {
-    const int16_t angle = TRI_TAIL_SERVO_ANGLE_MID + constrainedPIDOutput * tailServoMaxAngle / 1000;
-    return getServoValueAtAngle(servoConf, angle);
+    const int16_t angle = TRI_TAIL_SERVO_ANGLE_MID + constrainedPIDOutput * tailServo.maxAngle / 1000;
+    const uint16_t normalServoValue = getServoValueAtAngle(servoConf, angle);
+
+    return normalServoValue;
 }
 
 void triServoMixer(int16_t PIDoutput)
 {
+    static pt1Filter_t feedbackFilter;
+
     // Dynamic yaw expects input [-1000, 1000]
     PIDoutput = constrain(PIDoutput, -1000, 1000);
-    // Scale the PID output based on tail motor speed (thrust)
-    PIDoutput = dynamicYaw(PIDoutput);
-
-    static pt1Filter_t feedbackFilter;
-    if (gpMixerConfig->tri_servo_feedback != TRI_SERVO_FB_VIRTUAL)
-    {
+    PIDoutput = getScaledPIDatThrottle(PIDoutput);
+    if (gpMixerConfig->tri_servo_feedback != TRI_SERVO_FB_VIRTUAL) {
         // Read new servo feedback signal sample and run it through filter
-        tailServoADC = pt1FilterApply4(&feedbackFilter, adcGetChannel(tailServoADCChannel), 70, getdT());
+        tailServo.ADC = pt1FilterApply4(&feedbackFilter, adcGetChannel(tailServoADCChannel), 70, getdT());
     }
-
     // Dynamic yaw expects input [-1000, 1000]
     PIDoutput = constrain(PIDoutput, -1000, 1000);
-
     // Linear servo logic only in armed state
-    if (ARMING_FLAG(ARMED))
-    {
+    if (ARMING_FLAG(ARMED)) {
         // Scale the PID output based on tail motor speed (thrust)
-        PIDoutput = dynamicYaw(PIDoutput);
-
+        PIDoutput = getScaledPIDatThrottle(PIDoutput);
```

Comment:
Not related to this commit, but just noticed that this is the second call to the function. Probably shouldn't be here, might cause some nasty behavior.

Good to review own code once in a while :).